### PR TITLE
MAINT: Always use PyCapsule instead of PyCObject in mtrand.pyx

### PR DESCRIPTION
**Pull request by charris**
Python 2.7 has a backport of PyCapsule so we no longer need to support PyCObject. This PR makes that change and removes the no longer needed mt_compat.h file.

Some Deprecation warning are also hidden in the `numpy/random/tests`.

### CODE REVIEW

File: numpy/random/mtrand/mtrand.pyx [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 include "Python.pxi"
 include "numpy.pxd"
+include "cpython/pycapsule.pxd"
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: I'm curious to see if this path is portable to Windows. I suspect not and wonder why the path isn't in the normal cython search path.

Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: Looks like it isn't a problem.

Comment (njsmith) [inline numpy/random/mtrand/mtrand.pyx]: Yeah, windows is totally happy to accept `/` as a path separator.

**Comment by njsmith** [comment]
Looks fine to me. I'm a little puzzled about why we're using PyCapsule at all here... I guess just because it's easier to have a dict-of-Python-callables than to have a dict-of-cdef-function-pointers? Doesn't matter much I guess.

**Comment by charris** [comment]
> dict-of-Python-callables than to have a dict-of-cdef-function-pointers?

Yes, that was the problem, dict-of-cdef-function-pointers didn't work at all.

**Comment by charris** [comment]
I've toyed a bit with making the PyCapsule object public, but can't convince myself that it is a good idea.

**Comment by charris** [comment]
Strictly speaking, should probably check for error from `PyCapsule_GetPointer` also, but as long as everything is private should be OK to omit that.

**Comment by njsmith** [comment]
Yeah, let's not go adding the whole `rk_state` machinery to our public abi... We have enough trouble with our abi exposing too much already :-)

**Pull request merged**